### ISSUE TITLE
Support React v16.0

### ISSUE THREAD

**Issue by greenteamsite** [christianalfoni/formsy-react#481]
React v16.0 has been released with massive improvements.

**Comment by Kaishiyoku** [comment]
The blog post is saying that there shouldn't be any breaking changes when coming from 15.x.x (https://facebook.github.io/react/blog/2017/09/26/react-v16.0.html#upgrading)

**Comment by twisty** [comment]
Is this just a matter of adding ` || ^16.0.0` to the peerDependencies?

https://github.com/christianalfoni/formsy-react/blob/ebd84bb02613b355a02fd3e3cffb9d6dcd46b500/package.json#L47

That should be okay for the upcoming 1.0.x release, but the 0.x releases use mixins – are they supported in React 16?

**Comment by twisty** [comment]
The current release of the beta on npm has React v15 listed as a dependency. If you upgrade your app to React v16, two versions of React are loaded and chaos ensues.

https://github.com/christianalfoni/formsy-react/blob/998e2be44f4dcf3c49162244a4ef9e813af6f5c3/package.json

As this has already been fixed on the beta branch, is there any word on pushing a new release to npm?

**Comment by Jareechang** [comment]
whats the road map right now ? I would love to contribute and make updates but it seems pretty in-active.

![screen shot 2017-09-28 at 12 30 13 pm](https://user-images.githubusercontent.com/9851316/30986602-ddcaa9be-a448-11e7-872f-8364a569a4e0.png)

**Comment by kernel72** [comment]
There is another repo presents on Github - fork of this one. https://github.com/maccuaa/formsy-react-2. It looks like ready for React 16. AFAIK it was made because of low activity of this one repo. 
I have tried it in my projects, and looks like ok.

**Comment by Kaishiyoku** [comment]
Thanks for the link to the fork! Will look forward to it. It's sad that there hasn't happened much recently.

**Comment by aliogaili** [comment]
The other repo doesn't seem to work with [formsy-react-components ](https://github.com/twisty/formsy-react-components) I hope this repo get updated.

**Comment by twisty** [comment]
Hi @aliogaili, (Speaking as the maintainer of the `formsy-react-components package`). I'm not keen to support forks of `formsy-react` right now.

I agree the stuttered release pattern of `formsy-react` is frustrating, especially when we want to test with the latest shiny new React. But there is some nice stuff on the beta branch that seems like it could reboot this project, and I'd like to support that before switching to or otherwise supporting for forks.

Note: I'm not passing any judgment or expressing a preference about any forks here. It's just easier for consumers of `formsy-react-components` to couple with the `formsy-react` mothership.

**Comment by aliogaili** [comment]
Thanks @twisty, I also think that's the right path.

**Comment by maccuaa** [comment]
I agree, I would rather contribute to this project than have to maintain a fork.

**Comment by Azerothian** [comment]
I think i need to put it out there that react-formsy in its current state works fine with React 16.

**Comment by aliogaili** [comment]
I'm running into the following error when I update to React 16:

<img width="794" alt="screen shot 2017-10-06 at 7 17 25 am" src="https://user-images.githubusercontent.com/1675307/31261700-c1704ea8-aa66-11e7-80ad-149c592c4dc0.png">

Any idea on what might be causing this?

**Comment by aliogaili** [comment]
Ok I went to the node_modules/formsy-react/node_modules and deleted the react folder then the error went away! when I do npm install again, the folder gets recreated (with react react@15.6.2) but the error is no longer showing. 

Does anyone know what might be causing this behavior?

**Comment by twisty** [comment]
@aliogaili The reason react 15 is getting installed again is that the beta published to npm has react 15 listed as a **`dependency`**. It should be a **`peerDependency`**. This was fixed on the beta branch over a month ago, but there hasn't been a new beta release pushed to npm since the fix.

"I know!", you might think. "I'll just install it straight from the github repo!". Well, this does work, but takes a bit of twiddling. You may not get the built release files (the `lib` folder doesn't exist in the repo, and for some reason `yarn` wasn't transcoding them on install, YMMV):

This installs the latest version of the beta branch, transcodes the source, then removes any `devDependencies`.

```sh
yarn add https://github.com/christianalfoni/formsy-react#beta
cd ./node_modules/formsy-react
yarn install
yarn install --production --ignore-scripts
```

I think you'll still get a warning about the react peer dependency, but it should run okay.

**Comment by aliogaili** [comment]
If you download the tar referenced by the [resolved url](https://registry.npmjs.org/formsy-react/-/formsy-react-1.0.0.tgz) then you'll get a package.json that has react 15 referenced as a dependency (and not peer dependency). 

Moving react from dependency to peer dependency was done by the [last commit here](https://github.com/christianalfoni/formsy-react/commit/42e39cbc8fa14333374c56b38b9b9bf389916991) and it has not been published to npm  yet, which explains why I keep getting the older package.json with the react 15 as dependency. 

So how are you folks using the last commit?

**Comment by aliogaili** [comment]
Thank you so much @twisty ! that's what I'm looking for, it was a bit confusing for me, but now it's clear.

**Comment by aliogaili** [comment]
@twisty I've following the steps you gave and I keep getting `uncaughtException: Cannot find module 'formsy-react`

Formsy-react is in the package.json and in the node_modules folders but the app doesn't start and I receive this exception :(

**Comment by twisty** [comment]
Hi @aliogaili, I think technically it's all installed properly, but I think those errors are because the alpha 7 of `formsy-react-components` doesn't support the tip of the `formsy-react` beta branch. (It looks like `Formsy.Form` has gone away).

I've got an alpha ready to go that fixes this, but would prefer to wait for a new release from this repo.
 I'm not keen on documenting the process of depending on this branch, which won't be fun.

I'm hesitant to call out the batsignal for @aesopwolf or another maintainer, but it would really help to get a response or an ETA on this.

**Comment by aliogaili** [comment]
I just downloaded the version currently published on the NPM registry, patched it to remove the react dependency and installed it locally, it's working, for now.

But I really hope we can hear soon from the maintainer(s) of formsy-react and we get a newer version on NPM. This is a very useful and well designed library and really not much effort needs to get it up and running. I'm sure more people will run through this issue as they start updating to React 16, so the sooner this get resolved the better. Hopefully, the batsignal does it job :)

**Comment by aliogaili** [comment]
@Semigradsky I see you made contributions to formsy-react, and as it stand it's currently blocking the adoption of React 16 because [the last comment](https://github.com/christianalfoni/formsy-react/pull/473) has not been published to NPM.

@twisty there are two forks of this library due to the inactivity here:

1. [formsy-react-2 ](https://github.com/maccuaa/formsy-react-2), which has been created by the @maccuaa who is also maintaining [formsy-mui](https://github.com/maccuaa/formsy-mui)
2. [formsy-react-es6](https://github.com/track0x1/formsy-react/blob/v15-version/package.json) forked by @track0x1 who is also maintaining [formsy-react-native](https://github.com/track0x1/formsy-react-native) 

I think both of those forks exist due to the lack of activity here at the moment. I'm blocked from publishing my project because I'm using formsy-react-components which is depending on this version of formsy-react and it's conflicting with React 16. So can you please consider aligning with one of the two libraries (or even maintain your own fork if you wish) so we can move forward with this? 

Thanks!

**Comment by rkuykendall** [comment]
This issue has been moved to the new `formsy` organization: https://github.com/formsy/formsy-react/issues/7

**Comment by rkuykendall** [comment]
This issue has been **SOLVED** in the new `formsy` organization with Formsy `1.1.0`: https://github.com/formsy/formsy-react/
